entrypoint.sh

```diff
@@ -1,44 +1,115 @@
+    echo
+    echo "This is an internal error, please file an issue at https://github.com/returntocorp/semgrep/issues/new/choose"
+    echo "and include the log output from above."
+    exit $bento_result
+  fi
+
+  return $bento_result
 }
 
-function uses() {
-    [ ! -z "${1}" ]
+handle_pull_request() {
+  # the github ref would be `refs/pull/<pr #>/merge` which isn't known by name here
+  # the github sha seems to refer to the base on re-runs
+  # so we keep our own head ref around
+  real_head_sha=$(git rev-parse HEAD)
+
+  echo
+  echo "== [1/3] going to go back to the commit you based your pull request on…"
+  echo
+  git checkout "${GITHUB_BASE_REF}"
+  git status --branch --short
+
+  echo
+  echo "== [2/3] …now adding your pull request's changes back…"
+  echo
+
+  git checkout "${real_head_sha}" -- .
+  git status --branch --short
+
+  echo
+  echo "== [3/3] …and seeing if there are any new findings!"
+  echo
+  bento init &> /dev/null
+  bento check --tool=semgrep
 }
 
-function usesBoolean() {
-    [ ! -z "${1}" ] && [ "${1}" = "true" ]
+handle_push() {
+  echo
+  echo "== seeing if there are any findings"
+  echo
+  bento init &> /dev/null
+  bento check --all --tool=semgrep
 }
 
-function main() {
-    echo "" # see https://github.com/actions/toolkit/issues/168
-
-    if usesBoolean "${INPUT_ERROR}"; then
-        ERROR="--error"
-    else
-        ERROR=""
-    fi
-
-    if uses "${INPUT_OUTPUT}"; then
-        OUTPUT_ARG="--output ${INPUT_OUTPUT}"
-    else
-        OUTPUT_ARG=""
-    fi
-
-    set +e
-    OUTPUT=$(/bin/semgrep ${ERROR} --config "${INPUT_CONFIG}" ${OUTPUT_ARG} $INPUT_TARGETS)
-    EXIT_CODE=$?
-    set -e
-    ## echo to STDERR so output shows up in GH action UI
-    echo >&2 $OUTPUT
-    echo "::set-output name=output::${OUTPUT}"
-    exit $EXIT_CODE
+handle_unknown() {
+  echo "== [ERROR] the Semgrep action was triggered by an unsupported GitHub event."
+  echo
+  echo "This error is often caused by an unsupported value for `on:` in the action's configuration."
+  echo "To resolve this issue, please confirm that the `on:` key only contains values from the following list: [pull_request, push]."
+  echo "If the problem persists, please file an issue at https://github.com/returntocorp/semgrep/issues/new/choose"
+  exit 2
+}
+
+check_prerequisites() {
+  if ! [[ -v GITHUB_ACTIONS ]]
+  then
+    echo "== [WARNING] this script is designed to run via GitHub Actions"
+  fi
+
+  if ! [[ -v INPUT_CONFIG ]] && ! [[ -e .bento/semgrep.yml ]]
+  then
+    echo "== [WARNING] you didn't configure what rules semgrep should scan for."
+    echo 
+    echo "Please either set a config in the action's configuration according to"
+    echo "https://github.com/returntocorp/semgrep-action#configuration"
+    echo "or commit your own rules at the default path of .bento/semgrep.yml"
+  fi
+}
+
+main() {
+  echo "== action's environment: $($bento_path --version), $(python --version)"
```

Comment:
Can we symlink `bento_path` so it shows up as `semgrep`?

Reply:
What I did here instead was update it to print `semgrep/0.6.1, bento/0.12.0, python/3.8.1`. Is that okay?

Reply:
sure